src/api/cpp/nixl_types.h

```diff
@@ -62,6 +62,24 @@ typedef enum {
     NIXL_ERR_NOT_SUPPORTED = -9
 } nixl_status_t;
 
+/**
+ * @enum nixl_sync_t
+ * @brief An enumeration of supported synchronization modes for NIXL
+ */
+typedef enum {
+    NIXL_SYNC_NONE,
+#ifndef DISABLE_MT
+    NIXL_SYNC_STRICT,
+#endif
+} nixl_sync_t;
+
+#ifdef DISABLE_MT
```

Comment:
given that we probably want to ship thread-safe library eventually, i'd make only NIXL_LOCK_GUARD macro depend on DISABLE_MT (or ENABLE_MT) build config. Having readable code and stable API/ABI is IMO more important than avoiding the overheads of std::mutex member in nixl agent and extra parameters for slow-path functions.

Reply:
See my reply to @aranadive regarding this.

Please note that it is not a given that we can have a clear split between fast and slow paths and/or avoid all locking on the fast path.